src/engine.js

```diff
@@ -98,36 +63,71 @@ const editTaskDescription = (index, newDescription) => {
     saveTasksToLocalStorage();
     displayItemsByIndex();
   } else {
-    alert('Invalid task index.');
+    alert('please add a task to init app');
   }
 };
 
-// Event listeners for the checkboxes and delete buttons
-const attachEventListeners = () => {
+// Display tasks
+const displayItemsByIndex = () => {
+  listContainer.innerHTML = '';
+
   list.forEach((item) => {
-    const { index } = item;
-    const completedRadio = document.querySelector(`input[name="completed_${index}"]`);
+    const { description, completed, index } = item;
+
+    const listItem = document.createElement('div');
+    const descriptionSpan = document.createElement('span');
+    const completedRadio = document.createElement('input');
+    const edit = document.createElement('button');
+    const hr = document.createElement('hr');
+
+    edit.classList.add('edit-task');
+
+    listItem.classList.add('icon');
 
+    descriptionSpan.textContent = description;
+    completedRadio.type = 'checkbox';
+    completedRadio.name = `completed_${index}`;
+    completedRadio.checked = completed;
+    edit.textContent = 'Edit task';
+
+    // Handle checkbox change event
     completedRadio.addEventListener('change', () => {
       item.completed = completedRadio.checked;
       saveTasksToLocalStorage();
     });
+
+    edit.addEventListener('click', () => {
+      const newDescription = prompt('Enter the new description:');
+      if (newDescription !== '') {
+        editTaskDescription(index, newDescription);
+      } else {
+        alert('description must be at least two chars');
+      }
+    });
+
+    listItem.appendChild(completedRadio);
+    listItem.appendChild(edit);
+    listItem.appendChild(descriptionSpan);
+    listItem.appendChild(hr);
+
+    listContainer.appendChild(listItem);
+
+    document.getElementById('taskInput').value = '';
   });
 };
 
-const Delete = document.getElementById('clear');
-
-Delete.addEventListener('click', () => {
-  list = list.filter((item) => !item.completed);
-  saveTasksToLocalStorage();
-  displayItemsByIndex();
-});
+// Save and update local storage
+const saveTasksToLocalStorage = () => {
+  localStorage.setItem('todoList', JSON.stringify(list));
+  const filteredList = list.filter((task) => !task.completed);
+  localStorage.setItem('todoList', JSON.stringify(filteredList));
+};
```

Comment:
In the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=All%20changes%20to%20the%20To%20Do%20List%20should%20be%20saved%20in%20local%20storage.), it was mentionned that all changes to the to do list should be saved in the local storage, including changing the state of completed from `false` to `true`. 
When saving items to the local storage, only the ones with the completed state of `false` are being saved. This causes issue when the page is reloaded, because then the items which weren't deleted by the user, won't appear on the list.
Before reloading: 
![image](https://github.com/paulshaun-kdtk/WebPack_Linked-List/assets/93520178/3f990807-0b13-4612-8071-cfcd6447536d)
After reloading (without clicking on the `Clear all completed button`: 
![image](https://github.com/paulshaun-kdtk/WebPack_Linked-List/assets/93520178/a0f5d4b9-7c62-4bc9-afc0-358b628b9995)